ext/pdo/pdo_stmt.c

```diff
@@ -1141,11 +1086,13 @@ PHP_METHOD(PDOStatement, fetchObject)
 		PDO_HANDLE_STMT_ERR();
 		RETVAL_FALSE;
 	}
-	do_fetch_opt_finish(stmt, 1);
 
+	reset_old_ctor_args:
+	if (stmt->fetch.cls.ctor_args != NULL) {
+		zend_hash_release(stmt->fetch.cls.ctor_args);
```

Comment:
I wonder under what circumstances this could actually reach a refcount of 0, isn't GC_TRY_DELREF enough?

Reply:
Actually the old code didn't refcount, so why do you do refcounting now?

Reply:
Good question, I think last time I tried to refactor this I was hitting UAF, but I was probably doing something wrong. It doesn't seem to be needed indeed.